skate-plugin/src/main/kotlin/com/slack/sgp/intellij/featureflags/PsiElementExtensions.kt

```diff
@@ -16,34 +16,31 @@
 package com.slack.sgp.intellij.featureflags
 
 import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
 import com.intellij.psi.util.childrenOfType
-
-fun PsiElement.getFirstChild(elementType: String): PsiElement? {
-  return this.childrenOfType<PsiElement>().firstOrNull {
-    it.node.elementType.toString() == elementType
-  }
-}
-
-fun PsiElement.getChildren(elementType: String): List<PsiElement> {
-  return this.childrenOfType<PsiElement>().filter { it.node.elementType.toString() == elementType }
-}
-
-fun PsiElement.getAnnotationEntries(): List<PsiElement> {
-  return this.getFirstChild("MODIFIER_LIST")?.getChildren("ANNOTATION_ENTRY") ?: listOf()
-}
-
-fun PsiElement.getAnnotationName(): String? {
-  return this.getFirstChild("CONSTRUCTOR_CALLEE")?.text
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtAnnotationEntry
+import org.jetbrains.kotlin.psi.KtEnumEntry
+import org.jetbrains.uast.UExpression
+import org.jetbrains.uast.toUElement
+
+fun KtEnumEntry.hasAnnotation(name: String): Boolean {
+  return this.getAnnotationEntries().any { it.shortName?.asString() == name }
 }
 
-fun PsiElement.isEnum(): Boolean {
-  return this.node?.elementType?.toString() == "ENUM_ENTRY"
+fun KtEnumEntry.getAnnotation(name: String): KtAnnotationEntry? {
+  return this.getAnnotationEntries().firstOrNull { it.shortName?.asString() == name }
 }
 
-fun PsiElement.hasAnnotation(name: String): Boolean {
-  return this.getAnnotationEntries().any { it.getAnnotationName() == name }
+fun PsiElement.getEnumIdentifier(): LeafPsiElement? {
+  return this.childrenOfType<LeafPsiElement>().firstOrNull { it.elementType == KtTokens.IDENTIFIER }
```

Comment:
Is this necessary? I think `KtEnumEntry` just extents `KtClass` and you should be able to get the name like any other class.

Reply:
Getting name is easy as you pointed out but then for annotation there is range() which takes textRange, ASTNode or PsiElement. If I get a name then I will I will have to read a file to find a textRange as follow and pass as .range(symbol.textRange)

```
  private fun findTextRangeForFlag(psiFile: PsiFile, flag: String): TextRange? {
    val elements = PsiTreeUtil.findChildrenOfType(psiFile, PsiElement::class.java)
    return elements.firstOrNull { it.text == flag }?.textRange
  } 
```
  So I wanted to keep it simply and directly pass LeadPsiElement which is of type PsiElement as 
  .range(symbol.element as PsiElement)
  
  Thoughts?

Reply:
that's not keeping it simple as it's doing manual AST parsing. As I mentioned in a few other spots, please take a look at similar processing of enums we do in the slack-lints repo. Here's a good example https://github.com/slackhq/slack-lints/blob/059b7f88083089e3ab959e454fe296e435c5c53b/slack-lint-checks/src/main/java/slack/lint/MoshiUsageDetector.kt#L1018

Once you have the `UEnumConstant`, it has a simple `getName()` function on it that returns a string.

Reply:
[ec81f44](https://github.com/slackhq/slack-gradle-plugin/pull/576/commits/ec81f4417cd0409bed9ab368d3bdb606b12ea717)